x-pack/legacy/plugins/lens/public/editor_frame_plugin/editor_frame/editor_frame.test.tsx

```diff
@@ -452,7 +466,7 @@ describe('editor_frame', () => {
       mockDatasource2.getLayers.mockReturnValue(['second', 'third']);
       const instance = mount(
         <EditorFrame
```

Comment:
This test broke - `testDatasource2` should also provide an expression for two layers, but that's not happening somehow.

Reply:
I don't see the failure you're referring to after merging in the latest feature/lens branch

Reply:
Sorry, I was referring to the removed lines a bit below: https://github.com/elastic/kibana/pull/42031/files#diff-868dbc53720fc69e014610bbe0d3f281L510 and https://github.com/elastic/kibana/pull/42031/files#diff-868dbc53720fc69e014610bbe0d3f281L524 - that should still happen

Reply:
I found the cause here, the test was not alerting me to the real failure after I updated snapshots. I added a stronger test and changed the mocks so that this test does what it says